contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -6978,6 +6978,10 @@ bbf_ExecDropStmt(DropStmt *stmt)
 
 			if (!relation)
 				continue;
+			
+			/* Restrict dropping of extended stored procedures for non-superuser roles */
+			if (stmt->removeType == OBJECT_VIEW && !superuser())
+				check_restricted_object(address.objectId, OBJECT_VIEW);
```

Comment:
Not related to this but I think we should have OidIsValid check for objectId before this. Please refer to procedure scenario below.

Reply:
Will add a check in the follow up commit.